### ISSUE TITLE
Field type incorrect on filter path

### ISSUE THREAD

**Issue by notbdu** [xichen2020/eventdb#126]
For the following test data:
```
{"service":"testNamespace","@timestamp":"2019-01-22T13:33:42-08:00","st":true,"sid":{"foo":2,"bar":4},"tt":"active","tz":-6,"v":1.5}
{"service":"testNamespace","@timestamp":"2019-01-22T13:34:42-08:00","st":true,"sid":{"foo":2,"bar":4},"tt":"active","tz":-6,"v":1.5}
{"service":"testNamespace","@timestamp":"2019-01-22T13:35:42-08:00","st":false,"sid":{"foo":3,"bar":6},"tt":"inactive","tz":-6,"v":15}
{"service":"testNamespace","@timestamp":"2019-01-22T13:36:42-08:00","st":false,"sid":{"foo":3,"bar":6},"tt":"inactive","tz":-6,"v":15}
{"service":"testNamespace","@timestamp":"2019-01-22T13:37:42-08:00","st":false,"sid":{"foo":3,"bar":6},"tt":"inactive","tz":-6,"v":15}
```

Got the following error when attempting to filter against the `st` field:
`field [st] does not have values of type bool`

**Comment by notbdu** [comment]
No longer seeing this issue. Seeing a different issue documented here: #151

**Comment by notbdu** [comment]
Was able to repro this again by filtering on the same field that we're also grouping by.

**Comment by xichen2020** [comment]
Was this the same issue with using `unsafe.String` that got fixed in #134 ? If so can I close this?

**Comment by notbdu** [comment]
No, this was an issue w/ the fetched field being closed in `applyFilters`.

**Comment by xichen2020** [comment]
Fixed in #153
